### PR TITLE
FIX: Modified convert_forward_solution

### LINKED ISSUE #4368: convert_forward_solution
In mne.forward, there's a function called `_to_fixed_ori`. 
I expected that 
```
# use convert_forward_solution
forward_convert = convert_forward_solution(
    forward_ori, surf_ori=True, force_fixed=True, copy=True)
```
give the same result as
```
# use _to_fixed_ori
forward_fixed = convert_forward_solution(
    forward_ori, surf_ori=True, force_fixed=False, copy=True)
forward_fixed = _to_fixed_ori(forward_fixed)
```
Both expression should do the same, i.e., rotate to surf_ori and then convert to fixed.

The problem is that the results differ (`assert_allclose(forward_convert['sol']['data'], forward_fixed['sol']['data'])` fails).

So which way is recommended?
Should we get rid of one of the options (either by deleting `_to_fixed_ori`) or by calling `_to_fixed_ori` in `convert_forward_solution`?

### LINKED ISSUE #4393: bug? writing fixed fwd
As discussed with @agramfort the following code results in an error in `assert_allclose`:

```
import os.path as op
from numpy.testing import assert_allclose
from mne.datasets import testing
from mne import read_forward_solution, write_forward_solution
from mne.utils import _TempDir

data_path = testing.data_path(download=False)
fname_meeg_grad = op.join(data_path, 'MEG', 'sample',
                          'sample_audvis_trunc-meg-eeg-oct-2-grad-fwd.fif')
fwd_fixed = read_forward_solution(fname_meeg_grad, surf_ori=True,
                                  force_fixed=True)
temp_dir = _TempDir()
fname_temp = op.join(temp_dir, 'test-fwd.fif')
write_forward_solution(fname_temp, fwd_fixed, overwrite=True)
fwd_fixed_io = read_forward_solution(fname_temp)
assert_allclose(fwd_fixed['sol']['data'], fwd_fixed_io['sol']['data'])
```
In my opinion, the reason is line 749 in forward.py, which is `if fwd['surf_ori'] is True`, so it tries to rotate fixed fwd, too. Replacing this line with `if fwd['surf_ori'] is True and fwd['source_ori'] == FIFF.FIFFV_MNE_FREE_ORI:` fixes the problem.

If confirmed, I can add this bugfix to #4372.

### PR DESCRIPTION
**Pull request by larsoner**
Rebased version of #4372.

closes #4393, closes #4368

### CODE REVIEW

File: examples/inverse/plot_dics_beamformer.py [hunk 1]
```diff
@@ -53,7 +53,8 @@
 evoked = epochs.average()
 
 # Read forward operator
-forward = mne.read_forward_solution(fname_fwd, surf_ori=True)
+forward = mne.read_forward_solution(fname_fwd)
+forward = mne.convert_forward_solution(forward, surf_ori=True)
```
Comment (agramfort) [inline examples/inverse/plot_dics_beamformer.py]: isn't the conversion to surf ori now done automatically in the inverse solvers if 0 < loose < 1 ?

maybe we can skip a few of these conversion lines?

File: mne/forward/forward.py [hunk 2]
```diff
@@ -379,17 +379,17 @@ def _merge_meg_eeg_fwds(megfwd, eegfwd, verbose=None):
...
     ----------
     fname : string
         The file name, which should end with -fwd.fif or -fwd.fif.gz.
-    force_fixed : bool, optional (default False)
+    force_fixed : None | bool, optional (default None)
         Force fixed source orientation mode?
-    surf_ori : bool, optional (default False)
+    surf_ori : None | bool, optional (default None)
         Use surface-based source coordinate system? Note that force_fixed=True
         implies surf_ori=True.
```
Comment (agramfort) [inline mne/forward/forward.py]: mark these params are now deprecated in the docstring

File: mne/minimum_norm/inverse.py [hunk 3]
```diff
@@ -1294,6 +1293,9 @@ def make_inverse_operator(info, forward, noise_cov, loose='auto', depth=0.8,
         detected automatically. If int, the rank is specified for the MEG
         channels. A dictionary with entries 'eeg' and/or 'meg' can be used
         to specify the rank for each modality.
+    use_cps : None | bool (default None)
+        Whether to use cortical patch statistics to define normal
+        orientations. Only used when surf_ori and/or force_fixed are True.
```
Comment (agramfort) [inline mne/minimum_norm/inverse.py]: remove

Only used when surf_ori and/or force_fixed are True.

it's not params of this function

**Comment by agramfort** [comment]
@larsoner I am bit scared by all these branching in forward.py

how good is the coverage of forward.py ? do you think we have now an exhaustive non-regression benchmark with mne-c?

File: mne/minimum_norm/inverse.py (hunk 3 above)
Comment (larsoner) [reply mne/minimum_norm/inverse.py]: But you actually need it here. When doing depth weighting, you must provide a free orientation forward. So conversion happens later, within `make_inverse_operator`.

Comment (larsoner) [reply mne/minimum_norm/inverse.py]: (and, as you say, these conversions are now sometimes automatic, so whether or not CPS should be used needs to be a param in those cases, as well)

**Comment by larsoner** [comment]
> how good is the coverage of forward.py 

It's only 77%, but looking at the misses it's a lot of `raise`s for conditions that we never hit (i.e., bad files / sanity checks) and some code for `sol_grad`, which we never use in MNE (it has never been tested).

> do you think we have now an exhaustive non-regression benchmark with mne-c?

`sol_grad` aside, I thought we already had it :) But this has the CPS-based rotations that I think the inverse test I'm adding (and is currently failing) in #4618 needs here:

https://travis-ci.org/mne-tools/mne-python/jobs/280944321#L2627

You can see the `source_nn` is mismatched, and it's exactly the `source_nn` you get from using `fwd[2::3]` after rotating the XYZ-orientation fwd to surface orientation while using cortical patch statistics. So this PR should fix that one, making it even more exhaustive.

And in the meantime, looking at the diff, the existing `forward` tests aren't changed much, other than an occasional `use_cps=False`, indicating that the functionality from `master` is preserved.

File: mne/minimum_norm/inverse.py (hunk 3 above)
Comment (agramfort) [reply mne/minimum_norm/inverse.py]: what I mean is that make_inverse_operator does not have surf_ori or force_fixed parameters.
So the docstring is unclear.

**Comment by larsoner** [comment]
Comments addressed. I removed the conversions where possible, leaving the ones that raised errors for a future PR to simplify those functions (a `git grep convert_forward` from within `examples/` should make it easy to find the instances for the follow-up PR).

**Comment by agramfort** [comment]
"I thought we already had it :) "

I feel suddenly confident :)

more seriously I agree that our tests shoudl prevent clear regressions... I hope

File: mne/minimum_norm/inverse.py (hunk 3 above)
Comment (larsoner) [reply mne/minimum_norm/inverse.py]: Ahh I see, hopefully fixed in second commit below

**Comment by agramfort** [comment]
thx @larsoner

**Pull request merged**